src/app/guards/form/form.guard.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, QueryList, ViewChildren } from "@angular/core";
+import { AbstractControl } from "@angular/forms";
+import { CanDeactivate } from "@angular/router";
+import { FormComponent } from "src/app/component/shared/form/form.component";
+import { Constructor } from "src/app/helpers/advancedTypes";
+
+/**
+ * Add form checking to the component
+ * @param Base Class to extend
+ */
+export function WithFormCheck<T extends Constructor<{}>>(
+  Base: T = class {} as any
+) {
+  class Temporary extends Base {
```

Comment:
The name `Temporary` can show up in type annotations. Best to name it something better. Maybe `FormCheckingPageComponent1?